### PR TITLE
EIP-4747: Simplify EIP-161

### PR DESCRIPTION
**Pull request by petertdavies**
Currently doesn't have a number or an Ethereum magicians thread. I will create a Ethereum magicians thread as soon as I have an number.

### CODE REVIEW

**Review by lightclient — DISMISSED** [review]

File: EIPS/eip-simplify-161.md [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+If the transaction, `CALL` or `SELFDESTRUCT` is reverted, then the state clearing is also reverted. As a special case, in block 2675119 (tx `0xcf416c536ec1a19ed1fb89e4ec7ffb3cf73aa413b3aa9b77d60e4fd81a4296ba`), an empty call is made to an empty account at `RIPEMD160` (precompile `0x3`), the call fails out of gas, but the empty account is removed anyway.
+
+The deletion MAY happen immediately or be processed at the end of the transaction. A client may assume that `STATICCALL`s to empty accounts never occur and that `CALL`s to empty accounts in `STATICCALL` contexts never occur.
+
+After the start of Byzantium, the only state clearing events are the two listed below. Clients MAY implement post Byzantium state clearing by hardcoding those events.
+
+* In block 4457731 (tx `0x63791f962e13e6b01ec13d38a8ab66c87c2f5a1768276f866300d900cca808fe`), `0xd29DFe5aE95B5C067a91F472Dac0d9be6700A4A9` recieves a zero value `SELFDESTRUCT` and is cleared.
+
+* In block 14049881 (tx `0xf955834bfa097458a9cf6b719705a443d32e7f43f20b9b0294098c205b4bcc3d`), the following accounts recieve zero value calls and are cleared.
+    <details>
```
[the remark this answers is not in the document]
Comment (axic) [reply EIPS/eip-simplify-161.md:45]: These usual show up as collapsed, so an indication/explainer may be needed.

File: EIPS/eip-simplify-161.md [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+type: Standards Track
+category: Core
+created: 2022-02-02
+requires: 161
+---
+
+
+## Abstract
+
+EIP 161 is overly complex and has a number of edgecases that are poorly documented and tested. This EIP takes advantage of the complete removal of all remaining empty accounts in block 14,049,881 to clarify it and allows implementors to not implement various edgecases that never occured and are not possible in future.
```
Comment (axic) [inline EIPS/eip-simplify-161.md:17]:
```suggestion
EIP-161 is overly complex and has a number of edge cases that are poorly documented and tested. This EIP takes advantage of the complete removal of all remaining empty accounts in block 14,049,881 to clarify it, and allows implementors to not implement various edge cases that never occured and are not possible in the future.
```

The Oxford comma is a suggestion, the others to me seem like typos.

Additionally the Motivation section is missing and some of this text seem to belong to it.

File: EIPS/eip-simplify-161.md [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+In particular, this EIP permits implementors who do not wish to support historical blocks to not implement state clearing at all.
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+Retroactively replace EIP 161 with the following rules:
```
Comment (axic) [inline EIPS/eip-simplify-161.md:24]: Do you mean these rules apply from the same block EIP-161 did? Perhaps better to make it clear here.

File: EIPS/eip-simplify-161.md (hunk 2 above)
Comment (axic) [inline EIPS/eip-simplify-161.md:17]: It would be nice mentioning the transaction hash.

File: EIPS/eip-4747.md [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+type: Standards Track
+category: Core
+created: 2022-02-02
+requires: 161
+---
+
+
+## Abstract
+
+Simplify the definition of EIP-161, removing the requirement for implementors to support edge cases that are impossible on Mainnet.
```
Comment (axic) [inline EIPS/eip-4747.md:17]: At least the first occurrence must be a link:
```suggestion
Simplify the definition of [EIP-161](./eip-161.md), removing the requirement for implementors to support edge cases that are impossible on Mainnet.
```
```suggestion
Simplify the definition of EIP-161, removing the requirement for implementors to support edge cases that are impossible on Mainnet.
```

File: EIPS/eip-4747.md [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+## Motivation
+
+EIP-161 is overly complex and has a number of edge cases that are poorly documented and tested. This EIP takes advantage of the complete removal of all remaining empty accounts in block 14,049,881 (tx `0xf955834bfa097458a9cf6b719705a443d32e7f43f20b9b0294098c205b4bcc3d`) to clarify it, and allows implementors to not implement various edge cases that never occurred and are not possible in the future.
+
+In particular, this EIP permits implementors who do not wish to support historical blocks to not implement state clearing at all.
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+Retroactively replace EIP-161, starting from the beginning of Spurious Dragon, with the following rules:
```
Comment (axic) [inline EIPS/eip-4747.md:28]: I think a actual block number is more useful here:
```suggestion
Retroactively replace EIP-161, starting from block 1... (aka. the activation of Spurious Dragon), with the following rules:
```

File: EIPS/eip-4747.md [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+---
+
+
+## Abstract
+
+Simplify the definition of EIP-161, removing the requirement for implementors to support edge cases that are impossible on Mainnet.
+
+## Motivation
+
+EIP-161 is overly complex and has a number of edge cases that are poorly documented and tested. This EIP takes advantage of the complete removal of all remaining empty accounts in block 14,049,881 (tx `0xf955834bfa097458a9cf6b719705a443d32e7f43f20b9b0294098c205b4bcc3d`) to clarify it, and allows implementors to not implement various edge cases that never occurred and are not possible in the future.
```
Comment (axic) [inline EIPS/eip-4747.md:21]: This EIP uses different styles for block numbers: it is comma delimited here and in the rationale, while without delimiters elsewhere.

My impression is that most EIPs do not include commas in block numbers, but it is up to you.

File: EIPS/eip-4747.md [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+Ropsten had empty accounts seeded at genesis. They appear to have been cleared early in Ropsten's history before the Byzantium Hardfork. Ropsten has never been checked for edgecases occurring. All other Ethereum testnets have had EIP-161 from genesis.
+
+As a security precaution all empty accounts on Ethereum Classic have been cleared, but no checks for edgecases occurring have been done. Due to EIP-161's age the vast majority of EVM compatible networks have supported it from genesis.
+
+## Security considerations
+
+This EIP is only equivalent to EIP-161 on Mainnet if the following facts and must not be finalised until they have been independently verified:
+
+1. No empty accounts are ever touched and then reinstated in the same transaction.
+2. The transactions in the Appendix are the only state clearing transactions on Mainnet after the start of Byzantium.
```
Comment (axic) [inline EIPS/eip-4747.md:308]: Should include block number for Byzantium here, imho.

**Comment by MicahZoltu** [comment]
What is the reasoning for not just updating EIP-161 directly?

File: EIPS/eip-4747.md [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+
+### Other networks
+
+Ropsten had empty accounts seeded at genesis. They appear to have been cleared early in Ropsten's history before the Byzantium Hardfork. Ropsten has never been checked for edgecases occurring. All other Ethereum testnets have had EIP-161 from genesis.
+
+As a security precaution all empty accounts on Ethereum Classic have been cleared, but no checks for edgecases occurring have been done. Due to EIP-161's age the vast majority of EVM compatible networks have supported it from genesis.
+
+## Security considerations
+
+This EIP is only equivalent to EIP-161 on Mainnet if the following facts and must not be finalised until they have been independently verified:
```
Comment (MicahZoltu) [inline EIPS/eip-4747.md:305]:
```suggestion
This EIP is only equivalent to EIP-161 on Ethereum Mainnet if the following facts are true:
```
EIPs should be written in their final form, which is presumably *after* independent verification has occurred.  I recommend mentioning the need for independent verification in the discussions-to link.

**Comment by petertdavies** [comment]
> What is the reasoning for not just updating EIP-161 directly?

This is actually a substantial change to the consensus rules. It's just one that is justified by only being different to EIP-161 in circumstances that have never happened.

File: EIPS/eip-simplify-161.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (petertdavies) [reply EIPS/eip-simplify-161.md]: I tested this locally and its fine.

[the remark this answers is not in the document]
Comment (petertdavies) [reply EIPS/eip-simplify-161.md]: It will appear identically to the reference list in [EIP-779](https://eips.ethereum.org/EIPS/eip-779)

**Review by MicahZoltu — APPROVED** [review]

**Pull request merged**